File: rust/tests/feedback_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_feedback_create_feedback_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .feedback
        .create_feedback(
            &"project_id".to_string(),
            &CreateFeedbackRequest {
                kind: CreateFeedbackRequestKind::ThumbsUp,
                target: CreateFeedbackRequestTarget::Trace,
                target_id: "targetId".to_string(),
                issue_id: None,
                note: None,
                session_id: None,
                span_id: None,
                tag: None,
                trace_id: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/projects/project_id/feedback", None, 1)
        .await
        .unwrap();
}

File: ruby/test/wire/feedback_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class FeedbackWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_feedback_create_feedback_with_wiremock
    test_id = "feedback.create_feedback.0"

    @client.feedback.create_feedback(
      project_id: "project_id",
      kind: "thumbs_up",
      target: "trace",
      target_id: "targetId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "feedback.create_feedback.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/feedback",
      query_params: nil,
      expected: 1
    )
  end
end
